include/class_loader/class_loader.hpp

```diff
@@ -75,8 +75,10 @@ std::string systemLibraryFormat(const std::string & library_name);
  * @brief This class allows loading and unloading of dynamically linked libraries which contain class
  * definitions from which objects can be created/destroyed during runtime (i.e. class_loader).
  * Libraries loaded by a ClassLoader are only accessible within scope of that ClassLoader object.
+ * This class inherit from enable_shared_from_this which means we must used smart pointers,
+ * otherwise the code might work but it may run into a leak.
```

Comment:
@ahcorde nit: 
```suggestion
 * This class inherits from `std::enable_shared_from_this`. It should be instantiated using an `std::shared_ptr`.
 * Code will otherwise work but memory may be leaked.
```